web/lib/utils/index.js

```diff
@@ -64,6 +64,10 @@ export const genUpdates = (originalData, newData) => {
   return updates
 }
 
+/*
+ * The following functions originally come from the MIT licensed Vben project.
+ */
+
 export const hasNull = obj => {
```

Comment:
The `hasNull` function not copied from anywhere, copied functions are `deepMerge`, `setObjToUrlParams`.

Reply:
The tool I'm using suggests it was, perhaps it was copied from somewhere else?

Reply:
I unable to find where it might come from, but it it doe come from StackOverlow or anywhere else we need to know.

Reply:
This function is quite simple and universal.
I don't understand why there would be any copyright ownership involved.

Reply:
It doesn't matter how simple you think it is, copyright is still likely to apply.

Reply:
And if it was copied from somewhere whose license is not compatible with the Apache license, then we also have a problem.

Reply:
This function wasn't copied from anywhere.